src/include/txn.i

```diff
@@ -903,8 +903,8 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
 
     /* If there's no visible update in the update chain or ondisk, check the history store file. */
     if (F_ISSET(S2C(session), WT_CONN_HS_OPEN) && !F_ISSET(S2BT(session), WT_BTREE_HS))
-        WT_RET_NOTFOUND_OK(
-          __wt_find_hs_upd(session, key, recno, cbt->upd_value, false, &cbt->upd_value->buf));
+        WT_RET_NOTFOUND_OK(__wt_find_hs_upd(session, key, ((WT_CURSOR *)cbt)->value_format, recno,
```

Comment:
We use `cbt->iface`, not `(WT_CURSOR *)cbt`. We always prefer not to cast.

Reply:
OK, I see a lot of places we do the cast. I'll create a separate ticket to clean that.

Reply:
Created WT-6110